Lib/glyphsLib/__main__.py

```diff
@@ -14,48 +14,185 @@
+    # https://docs.python.org/2/library/os.html#os.rename, for Windows we have
+    # to first remove the target file if it exists.
+    designspace_filename = os.path.join(
+        options.output_dir,
+        os.path.basename(os.path.splitext(options.glyphs_file)[0]) +
+        ".designspace")
+    if designspace_path != designspace_filename:
+        if os.path.exists(designspace_filename):
+            os.remove(designspace_filename)
+        os.rename(designspace_path, designspace_filename)
+
+    return 0
+
+
+def ufo2glyphs(args=None):
+    """Convert one designspace file or one or more UFOs to a Glyphs file."""
+    import fontTools.designspaceLib
+    import defcon
 
-def main(args=None):
-    opt = parse_options(args)
-    if opt.glyphs is not None:
-        if opt.instances is None:
-            glyphsLib.build_masters(opt.glyphs, opt.masters)
+    parser = argparse.ArgumentParser(description=ufo2glyphs.__doc__)
+    parser.add_argument(
+        "--version",
+        action="version",
+        version="glyphsLib %s" % (glyphsLib.__version__))
+    parser.add_argument(
+        "designspace_file_or_UFOs",
+        nargs="+",
+        metavar="DESIGNSPACE_FILE_OR_UFOS",
+        help=("A single designspace file *or* one or more UFOs to convert to "
+              "a Glyphs file."),
+    )
+    parser.add_argument(
+        "--output-path",
+        help=("The path to write the Glyphs file to."),
+    )
+
+    group = parser.add_argument_group('Roundtripping between UFOs and Glyphs')
+    group.add_argument(
+        "--no-preserve-glyphsapp-metadata",
+        action="store_false",
+        help=("Skip preserving Glyphs metadata in master UFOs and designspace "
+              "file, which would be used to minimize differences when "
+              "roundtripping between Glyphs and UFOs."),
+    )
+    group.add_argument(
+        "--enable-last-change",
+        action="store_false",
+        help=("Store modification timestamp in glyphs. Unnecessary "
+              "when using Git."),
+    )
+    group.add_argument(
+        "--enable-automatic-alignment",
+        action="store_false",
+        help=("Enable automatic alignment of components in glyphs."),
+    )
+    options = parser.parse_args(args)
+
+    sources = options.designspace_file_or_UFOs
+    if len(sources) == 1 and sources[0].endswith(
+            ".designspace") and os.path.isfile(sources[0]):
+        designspace_file = sources[0]
+        designspace = fontTools.designspaceLib.DesignSpaceDocument()
+        designspace.read(designspace_file)
+        object_to_read = designspace
+    elif all(
+            source.endswith(".ufo") and os.path.isdir(source)
+            for source in sources):
+        ufos = [defcon.Font(source) for source in sources]
+        ufos.sort(
+            key=lambda ufo: [  # Order the masters by weight and width
+                ufo.info.openTypeOS2WeightClass, ufo.info.openTypeOS2WidthClass],
```

Comment:
```
In [11]: a=[1,2,3]

In [12]: a.sort(key=lambda ufo: [None, None])

In [13]: a
Out[13]: [1, 2, 3]
```

Reply:
it will fail if one of the two is not None...  😛 

```python
>>> from defcon import Font
>>> f1 = Font()
>>> f2 = Font()
>>> f1.info.openTypeOS2WeightClass = 400
>>> f1.info.openTypeOS2WidthClass = 5
>>> # f2.info.openTypeOS2WidthClass = 500
>>> f2.info.openTypeOS2WidthClass = 4
>>> lst = [f2, f1]
>>> lst.sort(key=lambda ufo: [ufo.info.openTypeOS2WeightClass, ufo.info.openTypeOS2WidthClass])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '<' not supported between instances of 'int' and 'NoneType'
>>> f2.info.openTypeOS2WeightClass = 500
>>> lst.sort(key=lambda ufo: [ufo.info.openTypeOS2WeightClass, ufo.info.openTypeOS2WidthClass])
>>> lst
[<defcon.objects.font.Font object at 0x10342a470>, <defcon.objects.font.Font object at 0x103432828>]
```

Reply:
just use `lambda ufo: [ufo.info.openTypeOS2WeightClass or 400, ufo.info.openTypeOS2WidthClass or 5]`